airbyte-integrations/connectors/source-blob/source_blob/source.py

```diff
@@ -0,0 +1,79 @@
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+from abc import ABC, abstractmethod
+from typing import Any, List, Mapping, Optional, Tuple
+from traceback import format_exc
+from fnmatch import fnmatch
+
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from .stream import IncrementalFileStreamS3
+
+
+class SourceBlob(AbstractSource, ABC):
+
+    @property
+    @abstractmethod
+    def stream_class(self) -> str:
+        """TODO docstring"""
+
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Optional[Any]]:
+        """
+        This method checks two things:
+            - That the credentials provided in config are valid for access.
+            - That the path provided in config is valid to be globbed.
+        """
+        found_a_file = False
+        try:
+            for filepath in self.stream_class.filepath_iterator(logger, config.get("provider")):
+                for path_pattern in config.get("path_patterns"):
+                    fnmatch(filepath, path_pattern)  # test that matching on the pattern doesn't error
+                    found_a_file = True
+                    break  # just find first file then break, no need to search all here
```

Comment:
The `if` check is missing here:

```suggestion
                    if fnmatch(filepath, path_pattern)
                        found_a_file = True
                        break
```

Reply:
at this point, we simply want to test that:

- the connection is valid
- the provided patterns are valid (and won't error) on fnmatch

For purpose of speed we can break as soon as we find one file as we satisfy the above tests, otherwise we could end up iterating here for a while in a case where many blobs exist but only few that match our pattern(s).
I've made a slight tweak to this block to achieve the above exactly so thanks for pointing out!